### PR TITLE
[Play] - Intro Splash Screen

### PR DESCRIPTION
**Pull request by drewjhart**
This PR acts as the base for the `JoinGame` flow that is as follows:

Splash Screen -> Enter Game Code -> Enter Player Name -> Select Avatar -> How to Play Carousel

Additionally, it adds the first screen, `SplashScreen.tsx` in that flow. The Figma reference is here:
image
https://www.figma.com/file/PadzU8GqyhFLCjIt7rQkpE/RightOn-Apps?node-id=13583-154467&t=nuq6BdMbpGPDYcaa-4

Code Organization: 
The main idea here is to create a `JoinGame` page that has a switch to flip between the different Join Game states. I'd rather do this than hoist the whole thing to `GameSessionContainer` because the states for Join Game are distinct from those in `GameSessionState.current.` `GameSessionState.current` just has `TEAMS.JOINING` and each of the Join Game states are a subset of that one state. I'd like to reflect this in our code for clarity.

PR Structure:
To be honest, I am reverse engineering this PR. It used to have every screen integrated and I've broken it down on a screen-by-screen basis for ease of review. The structure is now:

[[Play] - Intro Splash Screen and Join Game Flow Base](https://github.com/rightoneducation/righton-app/pull/586)
-> [[Play] - Enter Game Code Screen #588](https://github.com/rightoneducation/righton-app/pull/588)
->->  [[Play] - Enter Player Name Screen #589](https://github.com/rightoneducation/righton-app/pull/589)
->->->  [[Play] - Select Avatar Page #590](https://github.com/rightoneducation/righton-app/pull/590)
->->->-> [[Play] - How to Play Screen #591](https://github.com/rightoneducation/righton-app/pull/591)

_Note: this PR contains image files that aren't directly associated with the Splash Screen (it contains monster icon images for the Select Avatar page). This is because I've developed an object to handle these icons that's also shared by the footer container. I didn't want to undo and then reapply these changes so hopefully this isn't confusing. I've tagged where this occurs in each of the files with comments._

### CODE REVIEW

File: play/src/components/AnswerSelector.stories.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 import { ComponentStory, ComponentMeta } from '@storybook/react';
 import { ThemeProvider } from '@mui/material/styles';
 import AnswerSelector from './AnswerSelector';
-import AnswerState from '../lib/PlayModels';
+import { AnswerState } from '../lib/PlayModels';
```
Comment (drewjhart) [inline play/src/components/AnswerSelector.stories.tsx:5]: PlayModels has more than one model now. Can no longer use default export, typ for all such changes

File: play/src/components/FooterContent.tsx [hunk 2]
```diff
@@ -1,13 +1,8 @@
 import React from 'react';
 import { styled, useTheme } from '@mui/material/styles';
 import { Typography, Container } from '@mui/material';
+import { monsterMap } from '../lib/PlayModels';
```
Comment (drewjhart) [inline play/src/components/FooterContent.tsx:4]: These changes are in anticipation of [[Play] - Select Avatar Page #590](https://github.com/rightoneducation/righton-app/pull/590). They move the icon handling to lib

File: play/src/lib/PlayModels.tsx
Comment (drewjhart) [inline play/src/lib/PlayModels.tsx:1]: Includes new icon handling described above (as well as state required for JoinGame phases (where we don't have a GameSessionState)

File: play/src/img/HowToPlay1.svg
Comment (drewjhart) [inline play/src/img/HowToPlay1.svg:1]: I'm being lazy not removing these images until [[Play] - How to Play Screen #591](https://github.com/rightoneducation/righton-app/pull/591). It created a bunch of git issues with tracked/untracked files through the nested PR's, so I'm just making a note here and will avoid reverse engineering PR's like this in the future.

File: play/src/containers/GameSessionContainer.tsx [hunk 3]
```diff
@@ -5,21 +5,26 @@ import {
...
-    <GameInProgress
-      {...gameSession}
-      teamAvatar={teamAvatar}
-      teamId="2d609343-de50-4830-b65e-71eb72bb9bef"
-    />
+    // <GameInProgress
+    //   {...gameSession}
+    //   teamAvatar={teamAvatar}
+    //   teamId="2d609343-de50-4830-b65e-71eb72bb9bef"
+    // />
```
Comment (ryanabooth) [inline play/src/containers/GameSessionContainer.tsx:27]: You can delete code with no remorse whenever you're using git. Highly discourage commented code.

File: play/src/lib/PlayModels.tsx [hunk 4]
```diff
@@ -1,7 +1,42 @@
...
+  };
+}
+
+export const monsterMap: MonsterMap = {
+  0: { icon: Icon0, monster: Monster0 },
+  1: { icon: Icon1, monster: Monster1 },
+  2: { icon: Icon2, monster: Monster2 },
+  3: { icon: Icon3, monster: Monster3 },
+  4: { icon: Icon4, monster: Monster4 },
+  5: { icon: Icon5, monster: Monster5 },
```
Comment (ryanabooth) [inline play/src/lib/PlayModels.tsx:41]: neat

File: play/src/lib/styledcomponents/StyledComponents.tsx [hunk 5]
```diff
@@ -32,3 +31,53 @@ export const PaginationContainer = styled(Container)(({ theme }) => ({
...
+export const JoinGameBackgroundContainer = styled(Stack)(({ theme }) => ({
+  height: '100%',
+  position: 'fixed', // 100%, fixed to prevent sizing changes on mobile based on url bar etc
+  width: '100%',
+  background: `${theme.palette.primary.radialGradient} no-repeat`,
+  backgroundSize: `100%`,
+  backgroundPosition: 'center',
+  display: 'flex',
+  alignItems: 'center',
+}));
```
Comment (ryanabooth) [inline play/src/lib/styledcomponents/StyledComponents.tsx:83]: Cool!

File: play/src/pages/JoinGame.tsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+export default function JoinGame({ joinGameState }: JoinGameProps) {
+  const theme = useTheme();
+  const isMobileDevice = useMediaQuery(theme.breakpoints.down('sm'));
+  const [inputError, setInputError] = useState(true); // eslint-disable-line @typescript-eslint/no-unused-vars
+  const [gameCodeValue, setGameCodeValue] = useState('####');
+  const [playerFirstName, setPlayerFirstName] = useState('First Name');
+  const [playerLastName, setPlayerLastName] = useState('Last Name');
+  const [avatar, setAvatar] = useState(0); // eslint-disable-line @typescript-eslint/no-unused-vars
```
Comment (ryanabooth) [inline play/src/pages/JoinGame.tsx:18]: why include the unused vars?

**Review by ryanabooth — APPROVED** [review]

File: play/src/containers/GameSessionContainer.tsx (hunk 3 above)
Comment (drewjhart) [reply play/src/containers/GameSessionContainer.tsx:27]: Thanks for the reviews here, Ryan.

Good call, comment code removed.

File: play/src/pages/JoinGame.tsx (hunk 6 above)
Comment (drewjhart) [reply play/src/pages/JoinGame.tsx:18]: Laziness honestly. I know these are coming in subsequent screens. Good call though, they have been removed.

**Review by maniramezan — APPROVED** [review]

**Pull request merged**